service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerTest.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.HearingBooking;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(HearingBookingDetailsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class HearingBookingDetailsControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final String ERROR_MESSAGE = "Enter a future date";
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void shouldReturnAnErrorWhenHearingDateIsSetToYesterday() throws Exception {
```

Comment:
General comment, you don't need to really change it right now, but I would expect controller test to be simple, two scenarios happy path and unhappy path.

The validation tests should happen in the service on the service test level, it probably means that the controller is doing validation which is not correct. I think it's simple validator so it's relatively ok.

Reply:
👍

Reply:
Yeah it would be nice to extract the validate method in your controller to a separate service. But with sprint end today, I think this is fine for now. We will take on tech debt. It is this way in other places also